tests/test_views.py

```diff
@@ -0,0 +1,143 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer, Car, Driver
+
+MANUFACTURER_URL = reverse("taxi:manufacturer-list")
+CAR_URL = reverse("taxi:car-list")
+DRIVER_URL = reverse("taxi:driver-list")
+
+
+class PublicManufacturerTest(TestCase):
+    def test_manufacturer_login_required(self):
+        response = self.client.get(MANUFACTURER_URL)
+
+        self.assertNotEqual(response, 200)
+
+
+class PrivateManufacturerTest(TestCase):
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            username="test1",
+            password="test1234567",
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturer(self):
+        Manufacturer.objects.create(name="test1")
+        Manufacturer.objects.create(name="test2")
+
+        response = self.client.get(MANUFACTURER_URL)
+        manufacturers = Manufacturer.objects.all()
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            list(response.context["manufacturer_list"]),
+            list(manufacturers)
+        )
+        self.assertTemplateUsed(response, "taxi/manufacturer_list.html")
+
+    def test_manufacturer_search(self):
+        Manufacturer.objects.create(name="BMW", country="test2")
+        filtered_manufacturer = Manufacturer.objects.filter(

```

Comment:
do u use it somewhere ?
also this test doesn't check searched page context 
u are only checking status code + template using, check manufacturers filtered list

Reply:
the same for other tests
also u can create manufacturers for all tests using setUp